apis/Google.Devtools.AspNet/Google.Devtools.AspNet/CloudTrace/TracerManager.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Web;
+
+namespace Google.Devtools.AspNet
+{
+    /// <summary>
+    /// Manages access to the current <see cref="IManagedTracer"/>.
+    /// </summary>
+    internal static class TracerManager
+    {
+        // The key to save the tracer under.
+        private const string TraceKey = "Google.Devtools.AspNet.CloudTrace";
+
+        /// <summary>
+        /// Sets the current <see cref="IManagedTracer"/>
+        /// </summary>
+        public static void SetCurrentTracer(IManagedTracer tracer)
+        {
+            HttpContext.Current.Items[TraceKey] = tracer;
+        }
+
+        /// <summary>
+        /// Gets the current <see cref="IManagedTracer"/> or null if non exists.
+        /// </summary>
+        public static IManagedTracer GetCurrentTracer()
+        {
+            return (IManagedTracer) HttpContext.Current.Items[TraceKey];
```

Comment:
I don't know enough about HttpContext to know whether that's appropriate for this. Are you happy that it is, or should I research a bit more?